exotica_python/src/Exotica.cpp

```diff
@@ -1052,15 +1093,66 @@ PYBIND11_MODULE(_pyexotica, module)
         return vec;
     });
 
+    ////////////////////////////////////////////////////////////////////////////
+    /// Shapes
+
+    // shape base class
+    py::class_<shapes::Shape, shapes::ShapePtr>(module, "Shape")
+        .def("scale", &shapes::Shape::scale)
+        .def("padd", &shapes::Shape::padd)
+        .def("scaleAndPadd", &shapes::Shape::scaleAndPadd)
+        .def("isFixed", &shapes::Shape::isFixed)
+        .def_readwrite("type", &shapes::Shape::type);
```

Comment:
Do we want this to be read-write?

Reply:
I mirrored the access rights from the original C++ interface class (`type` is `public`).
But I also do not see from the where changing the type of a derived `Shape` (`Sphere`, `Cylinder`, ...) might be useful.
I am happy to change it to read-only for the python binding if you wish.

Reply:
Yes, please change it to read-only. I dont think changing the shape would have an effect such as casting so it is prone to user error.